driver-mapping/src/main/java/com/datastax/driver/mapping/Mapper.java

```diff
@@ -932,19 +924,65 @@ public static Option saveNullFields(boolean enabled) {
             return new SaveNullFields(enabled);
         }
 
-        public Type getType() {
-            return this.type;
+        /**
+         * Creates a new Option object to specify whether an IF NOT EXISTS clause should be included in
+         * insert queries. This option is valid only for save operations.
+         * <p/>
+         * If this option is not specified, it defaults to {@code false} (IF NOT EXISTS statements are not used).
+         *
+         * @param enabled whether to include an IF NOT EXISTS clause in queries.
+         * @return the option.
+         */
+        public static Option ifNotExists(boolean enabled) {
+            return new IfNotExists(enabled);
+        }
+
+        final Type type;
+
+        protected Option(Type type) {
+            this.type = type;
         }
 
-        abstract void appendTo(Insert.Options usings);
+        /**
+         * Checks that the option is allowed for a given query type, in a given manager.
+         *
+         * @throws IllegalArgumentException if the option is not valid
+         */
+        abstract void validate(QueryType qt, MappingManager manager);
+
+        /**
+         * Whether the option has any impact on the prepared query string.
+         */
+        abstract boolean modifiesQueryString();
 
-        abstract void appendTo(Delete.Options usings);
+        /**
+         * Appends the option to the prepared statement associated to the mapper operation
+         * (might be a no-op, not all options affect the query string).
+         */
+        abstract void modifyQueryString(BuiltStatement query);
 
-        abstract void addToPreparedStatement(BoundStatement bs, int i);
+        /**
+         * "Binds" the option before the request gets executed (might be a no-op).
+         * <p/>
+         * If the option's value is actually set as a bound variable, then this must be done at
+         * the given index, and the method should return the new position.
+         * <p/>
+         * Options may also affect the statement itself, not a bound variable; in that case, the
+         * method must return the index unchanged.
+         */
+        abstract int bind(BoundStatement bs, int currentIndex);
```

Comment:
`bind` sounds a bit ambiguous to me since the implementations are going to either bind a parameter, or set an option on the statement. Couldn't we call it something like `applyOption`?